### DISCUSSION TITLE
Upstream prematurely closed connection while reading response header from upstream

### DISCUSSION THREAD

**Discussion by shettyh** [Kong/kong#10085] [Help]
Hi,

After upgrading to Kong 2.8.3, we are seeing increased number of 502 with error `Upstream prematurely closed connection while reading response header from upstream`. It is happening to only a php upstream that is behind apache server. Any config that can be tuned to avoid these failures ?

Thanks

**Comment by flrgh** [comment]
Hi @shettyh 

> `Upstream prematurely closed connection while reading response header from upstream`

Generally speaking, this means that Kong successfully proxied the request from the client to the upstream, but the upstream closed the TCP connection without sending any response. As such there's not a lot of debugging that can be done from the perspective of the proxy. Given that Kong was at least able to _send_ a request, there should be some record on the upstream of this (assuming that it's going to the correct destination). Have you cross-referenced these events with the Apache/PHP logs?

> After upgrading to Kong 2.8.3

Can you let us know which version of Kong you were running prior to 2.8.3? And were any other changes made to your environment/topology along with the Kong upgrade?

**Comment by ghost** [comment]
Usually this is related to the timeout configuration of the upstream, you can check it and try to turn it up.
